### PR TITLE
Add an introduction

### PR DESCRIPTION
**Pull request by Swiddis**
The current cheatsheet goes into the commands and what they do without really introducing why one would want to do them. This PR does a few things:
- Adds an introduction to explain what Git is, some core terminology, and why we even care. This is placed in a new `instruction` directory.
- Also in that directory is the anatomy of a repository: what things there are and what they do.
- Fixes [Nested lists not rendering properly in GitHub preview](https://github.com/josephreed2600/git-cheatsheet/issues/1)
- Adds a new log command to `10-stage-commit.md` that I use often in debugging.

### CODE REVIEW

**Comment by josephreed2600** [comment]
hey google how do i split up a pull request github

**Comment by josephreed2600** [comment]
Per side-channel discussion, it sounds like Swiddis@ may interested in contributing more instructional and explanatory material. I'm all for it---this is a great opportunity to shed Yet Another light on a daunting but crucial topic for developing developers. My suggested change would be to split up the cheat sheet from the instruction, creating two "paths": think compiling into two documents, a book and a quick reference. This would allow the instructional path to be more verbose, cover more cases, and explain in more depth, while the quick reference path is just that.

In short, action items for this PR:

 1. Create two top-level dirs, e.g. `reference` and `instruction`
 2. Move the existing content into the reference dir
 3. Add the new introduction to the instruction dir
 4. Either add a Makefile to build the instructional content similar to the existing one, or adapt the Makefile to build both paths

I'll admit writing Makefiles was kind of a side-thing-I-wanted-to-learn that I shoehorned into this project because normal people like PDFs and I like Markdown and the build process was getting out of hand and yeah.

**Comment by Swiddis** [comment]
I may have updated overzealously. My bad.

**Comment by Swiddis** [comment]
I don't mean to rush you, but it's been 6 months, any updates on that review? Is my writing that bad?
